app/models/user.rb

```diff
@@ -3,4 +3,19 @@
 class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
+
+  has_one_attached :avatar do |attachable|
+    attachable.variant :thumb, resize_to_limit: [100, 100]
+  end
+
+  validate :correct_avatar_mime_type
+
+  private
+
+  def correct_avatar_mime_type
+    return unless avatar.attached? && !avatar.content_type.in?(%w[image/jpeg image/png image/gif])
+
+    avatar.purge
+    errors.add(:avatar, message: I18n.t('errors.messages.wrong_format'))
```

Comment:
```suggestion
    errors.add(:avatar, :wrong_format)
```

のようにエラーのキーを与えてそこから翻訳を取得する方法もあります。
このやり方もやってみてください〜。

Reply:
そのような方法でエラーメッセージの翻訳ができるとは知りませんでした。
Railsガイドを調べてみると、[4.5.1 エラーメッセージのスコープ](https://railsguides.jp/i18n.html#%E3%82%A8%E3%83%A9%E3%83%BC%E3%83%A1%E3%83%83%E3%82%BB%E3%83%BC%E3%82%B8%E3%81%AE%E3%82%B9%E3%82%B3%E3%83%BC%E3%83%97)に説明がありますね。
勉強になりました🙇‍♂️